### PR TITLE
Reduce default values of the quest file

### PR DESCRIPTION
**Pull request by KatatsumuriPan**
To reduce the quest file size, omit default values.

- Renamed NBTReplaceUtil -> NBTUtil
- Added "reduce" flag parameter to writeToNBT()s to specify whether to reduce or not. (Don't reduce on nbt editor.)
- Added IPropertyReducible to omit default values.

### CODE REVIEW

**Comment by KatatsumuriPan** [comment]
I'll fix backward compatibility

**Comment by Rongmario** [comment]
Please fix the merge conflict

File: src/main/java/betterquesting/NBTReplaceUtil.java [hunk 1]
```diff
@@ -1,34 +0,0 @@
-package betterquesting;
-
-import net.minecraft.nbt.NBTBase;
-import net.minecraft.nbt.NBTTagCompound;
-import net.minecraft.nbt.NBTTagList;
-import net.minecraft.nbt.NBTTagString;
-
-public class NBTReplaceUtil {
-    @SuppressWarnings("unchecked")
-    public static <T extends NBTBase> T replaceStrings(T baseTag, String key, String replace) {
```
Comment (Rongmario) [inline src/main/java/betterquesting/NBTReplaceUtil.java:10]: This class should stay to keep backwards compat, you can redirect to NBTUtil

File: src/main/java/betterquesting/storage/PropertyContainer.java [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
+import java.util.Objects;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
 
 public class PropertyContainer implements IPropertyContainer, INBTSaveLoad<NBTTagCompound> {
     private final NBTTagCompound nbtInfo = new NBTTagCompound();
+    // For reducing nbt
+    // To hold nbt values if the properties are not used (ex: the addon is temporarily removed), we cache and use only used properties to reduce nbt.
+    private final BiMap<String, IPropertyType<?>> id2PropertyMap = HashBiMap.create(); //ResourceLocation -> property
```
Comment (Rongmario) [inline src/main/java/betterquesting/storage/PropertyContainer.java]: This can just have `ResourceLocation` as the key.

**Review by Rongmario — CHANGES_REQUESTED** [review]

**Review by Rongmario — APPROVED** [review]
Pulling this in after #115.

**Comment by Rongmario** [comment]
Superseded by #117, thanks for the contribution!

**Pull request closed without merging**